cmd/infra/aws/iam.go

```diff
@@ -320,109 +318,66 @@ func (o *CreateIAMOptions) CreateOIDCResources(iamClient iamiface.IAMAPI, s3Clie
 		Bytes:   x509.MarshalPKCS1PrivateKey(privKey),
 	})
 
-	pubKey := &privKey.PublicKey
-	pubKeyDERBytes, err := x509.MarshalPKIXPublicKey(pubKey)
+	issuerURLWithProto := o.IssuerURL
+	url, err := url.Parse(issuerURLWithProto)
 	if err != nil {
+		log.Error(err, "Failed to parse issuer-url", "issuer-url", issuerURLWithProto)
 		return nil, err
 	}
-
-	hasher := crypto.SHA256.New()
-	hasher.Write(pubKeyDERBytes)
-	pubKeyDERHash := hasher.Sum(nil)
-	kid := base64.RawURLEncoding.EncodeToString(pubKeyDERHash)
-
-	var keys []jose.JSONWebKey
-	keys = append(keys, jose.JSONWebKey{
-		Key:       pubKey,
-		KeyID:     kid,
-		Algorithm: string(jose.RS256),
-		Use:       "sig",
-	})
-
-	jwks, err := json.MarshalIndent(KeyResponse{Keys: keys}, "", "  ")
-	if err != nil {
+	if url.Scheme != "https" {
+		err = fmt.Errorf("issuer-url must be https")
+		log.Error(err, "invalid issuer-url", "issuer-url", issuerURLWithProto)
```

Comment:
This is both fatal and can be computed during very early flag validation, so could it be moved way up to validation before we even attempt to do any work?